requirements.txt

```diff
@@ -1,6 +1,197 @@
+fsspec==2022.3.0
+future==0.18.2
+gast==0.5.3
+globus-sdk==3.3.1
+google-auth==2.6.0
+google-auth-oauthlib==0.4.6
+google-pasta==0.2.0
+grpcio==1.43.0
+h5py==3.2.1
+hmmlearn==0.2.6
+idna==2.8
+imageio==2.14.1
+imglyb==2.0.0
+importlib-metadata==4.10.1
+importlib-resources==5.4.0
+inflect==2.1.0
+iniconfig==1.1.1
+ipaddress==1.0.22
+ipykernel==6.9.1
+ipython==8.1.1
+ipython-genutils==0.2.0
+javabridge==1.0.19
+jedi==0.18.1
+jgo==1.0.3
+Jinja2==3.0.3
+jmespath==0.9.3
+joblib==1.1.0
+JPype1==1.3.0
+json5==0.9.6
+jsonschema==4.4.0
+jupyter-client==7.1.2
+jupyter-core==4.9.2
+jupyter-server==1.13.5
+jupyterlab==3.3.0
+jupyterlab-pygments==0.1.2
+jupyterlab-server==2.10.3
+keras==2.8.0
+Keras-Preprocessing==1.1.2
+kiwisolver==1.0.1
+libclang==13.0.0
+libsubmit @ git+https://github.com/MYPei/libsubmit.git@dd93180873010710917ed8ad6bfc03b64b8836a1
+locket==0.2.1
+lxml==4.7.1
+mahotas==1.4.5
+Markdown==3.3.6
+MarkupSafe==2.0.1
+matplotlib==3.1.3
+matplotlib-inline==0.1.3
+microscopeimagequality==0.1.0.dev5
+mistune==0.8.4
+mysqlclient==1.4.6
+nbclassic==0.3.6
+nbclient==0.5.12
+nbconvert==6.4.2
+nbformat==5.1.3
+nbstripout==0.5.0
+nest-asyncio==1.5.4
+networkx==2.2
+nose==1.3.7
+notebook==6.4.8
+notebook-shim==0.1.0
+numpy==1.22.2
+oauthlib==3.2.0
+opt-einsum==3.3.0
+packaging==21.3
+pandas==1.4.0
+pandocfilters==1.5.0
+paramiko==2.4.2
+parsl==1.1.0
+parso==0.8.3
+partd==1.2.0
+pathlib==1.0.1
+pickleshare==0.7.5
+Pillow==9.0.1
+pluggy==1.0.0
+prokaryote==2.4.4
+prometheus-client==0.13.1
+prompt-toolkit==3.0.28
+protobuf==3.19.4
+psutil==5.9.0
+pure-eval==0.2.2
+py==1.11.0
+pyasn1==0.4.8
+pyasn1-modules==0.2.8
+pycparser==2.19
+Pygments==2.11.2
+pyimagej==1.0.2
+PyJWT==2.3.0
+PyNaCl==1.3.0
+pyparsing==2.3.1
+pyrsistent==0.18.1
+PySide2==5.15.2.1
+pytest==6.2.5
+pytest-cov==3.0.0
+python-bioformats==4.0.5
+python-box==3.2.4
+python-dateutil==2.8.2
+python-javabridge==4.0.3
+pytz==2021.3
+PyWavelets==1.2.0
+pywin32==303
+pywinpty==1.1.6
+PyYAML==5.4.1
+pyzmq==18.0.1
+raven==6.10.0
+requests==2.27.1
+requests-oauthlib==1.3.1
+rsa==4.8
+runipy==0.1.5
+Rynner @ git+https://github.com/sa2c/rynner.git@e895b231bec89154929388e10289d9abe1947059
+s3transfer==0.5.1
+scikit-image==0.19.1
+scikit-learn==0.20.2
+scipy==1.7.3
+scyjava==1.4.1
+Send2Trash==1.8.0
+sentry-sdk==0.18.0
+shiboken2==5.15.2.1
+six==1.12.0
+sniffio==1.2.0
+stack-data==0.2.0
+tabulate==0.8.9
+tblib==1.7.0
+tensorboard==2.8.0
+tensorboard-data-server==0.6.1
+tensorboard-plugin-wit==1.8.1
+tensorflow==2.8.0
+tensorflow-io-gcs-filesystem==0.24.0
+termcolor==1.1.0
+terminado==0.13.3
+testpath==0.6.0
+tf-estimator-nightly==2.8.0.dev2021122109
+threadpoolctl==3.1.0
+tifffile==2022.2.2
+timeago==1.0.9
+toml==0.10.2
+tomli==2.0.1
+toolz==0.9.0
+tornado==6.1
+traitlets==5.1.1
+typeguard==2.13.3
+typing_extensions==4.0.1
+urllib3==1.26.8
+wcwidth==0.2.5
+webencodings==0.5.1
+websocket-client==1.3.1
+Werkzeug==2.0.2
+wrapt==1.13.3
+wxPython==4.1.0
+xarray==0.21.1
+zipp==3.7.0
+zope.interface==5.4.0
```

Comment:
This is more specific than is necessary. Can we trim this down to just the essentials, and ideally to a range of versions rather than an exact pinned version, similarly to how things are pre-change?